File: packages/case-boundary/src/boundary/ITriggerFunction.ts
import { BoundaryResult } from './Result';

/**
 * Interface called by ContractCase to invoke user's test code.
 */
export interface ITriggerFunction {
  /**
   * Trigger the user's test code and test the response from it.
   *
   * @returns BoundarySuccess when successful; BoundaryFailure with
   * kind=BoundaryFailureKindConstants.CASE_TRIGGER_ERROR if the trigger fails;
   * BoundaryFailure with
   * kind=BoundaryFailureKindConstants.CASE_VERIFY_RETURN_ERROR if the
   * verification function fails.
   */
  trigger(config: Record<string, unknown>): Promise<BoundaryResult>;
}

File: packages/dsl-java/src/main/java/io/contract_testing/contractcase/edge/ITriggerFunction.java
package io.contract_testing.contractcase.edge;

import org.jetbrains.annotations.NotNull;

public interface ITriggerFunction {

  /**
   * Trigger the user's test code and test the response from it.
   * <p>
   *
   * @param setupInfo This parameter is required.
   * @return BoundarySuccess when successful; BoundaryFailure with
   * kind=BoundaryFailureKindConstants.CASE_TRIGGER_ERROR if the trigger fails; BoundaryFailure with
   * kind=BoundaryFailureKindConstants.CASE_VERIFY_RETURN_ERROR if the verification function fails.
   */

  ConnectorResult trigger(final @NotNull ConnectorSetupInfo setupInfo);

}
